addons/onboarding/models/onboarding_onboarding.py

```diff
@@ -71,19 +77,20 @@ def action_close(self):
     def action_toggle_visibility(self):
         self.current_progress_id.action_toggle_visibility()
 
-    def write(self, values):
-        onboardings_per_company_update = self.browse()
-        if 'is_per_company' in values:
-            onboardings_per_company_update = self.filtered(
-                lambda onboarding: onboarding.is_per_company != values['is_per_company'])
+    def write(self, vals):
+        onboardings_changing_is_per_company = self.browse() if 'is_per_company' not in vals \
+            else self.filtered(lambda onb: onb.is_per_company != vals['is_per_company'])
 
-        res = super().write(values)
+        res = super().write(vals)
+
+        if onboardings_changing_is_per_company:
+            if onboardings_changing_is_per_company.step_ids:  # checked after super().write() in case they are removed.
```

Comment:
Looks like a lot of complicated code ... can't we use computed field to synchronize the is_per_company value ?

Reply:
I guess we can :)